src/lib.rs

```diff
@@ -627,10 +627,14 @@ bsize: BlockSize, bo: &BlockOffset) -> f64 {
     }
 
     let bs = mi_size_wide[bsize as usize];
-    let must_split = bo.x + bs as usize > fi.w_in_b || bo.y + bs as usize > fi.h_in_b;
 
-    let is_splittable = bsize > fi.min_partition_size;
-    let is_codable_unsplit = bsize < BlockSize::BLOCK_64X64 && !must_split;
+    // Always split if the current partition is too large
+    let must_split = bo.x + bs as usize > fi.w_in_b ||
+        bo.y + bs as usize > fi.h_in_b ||
+        bsize >= BlockSize::BLOCK_64X64;
+
+    // must_split overrides the minimum partition size when applicable
+    let can_split = bsize > fi.min_partition_size || must_split;
```

Comment:
And even this exact name 'can_split', which I was thinking it might be better than is_splittable!